samples/subsys/ipc/ipc_service/icmsg/remote/src/main.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+
+#include <ipc/ipc_service.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(remote, LOG_LEVEL_INF);
+
+#define MLEN_0 40
+
+struct data_packet {
+	unsigned char message;
+	unsigned char data[100];
+};
+
+
+K_SEM_DEFINE(bound_sem, 0, 1);
+
+static void ep_bound(void *priv)
+{
+	k_sem_give(&bound_sem);
+	LOG_INF("Ep bounded");
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	char received_data = *((char *)data);
+	static char expected_message = 'a';
+	static uint16_t expected_len = MLEN_0;
+
+	static unsigned long long cnt;
+	static unsigned int stats_every;
+	static uint32_t start;
+	static unsigned int err;
+
+	if (start == 0) {
+		start = k_uptime_get_32();
+	}
+
+	if (received_data != expected_message) {
+		err++;
+	}
+
+	if (len != expected_len) {
+		err++;
+	}
+
+	expected_message++;
+	expected_len++;
+
+	cnt += len;
+
+	if (expected_message > 'z') {
+		if (stats_every++ > 50) {
+			printk("%llu\n", (1000*cnt)/(k_uptime_get_32() - start));
```

Comment:
a little description of this number would be good. You may consider to use logging here instead of printk since it is in handler (but i'm not sure if that is necessary).

Reply:
> I can not use logger here as this must be as fast as possible

To be more precise: sample is never reaching lower priority thread because of high throughput (100% cpu load) so logging would not be able to handle messages in deferred mode (and immediate mode would be heavier than printk).

Reply:
I will add this.